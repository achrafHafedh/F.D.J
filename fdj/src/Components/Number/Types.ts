export type NumberProps = {
  index: number;
  value: number;
  onClick: Function;
  className: string;
};
