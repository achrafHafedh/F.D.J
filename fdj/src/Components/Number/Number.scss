.container-number {
  position: relative;
  display: inline-block;
  margin: 5px;

  .number {
    border-radius: 50%;
    background: white;
    color: black;
    font-size: 1rem;
    width: 25px;
    height: 25px;
  }

  input:checked {
    ~ .number {
      background: #263e83;
      color: white;
    }
    ~ label {
      background: #263e83;
      color: white;
    }
  }

  label {
    position: absolute;
    cursor: pointer;
    left: 4px;
    top: 4px;
    color: #263e83;
    font-weight: bold;
    font-size: 0.8rem;
    border-radius: 50%;
    &.small {
      padding: 0 5px;
    }
  }

  .visually-hidden {
    position: absolute;
    left: -100vw;
  }
}
