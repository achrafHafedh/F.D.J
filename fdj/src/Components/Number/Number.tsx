import { FormEvent, FunctionComponent } from "react";
import { NumberProps } from "./Types";

import "./Number.scss";

const Number: FunctionComponent<NumberProps> = ({ index, value, onClick, className }) => {
  return (
    <div className="container-number">
      <input
        type="checkbox"
        onChange={(v: FormEvent<HTMLInputElement>) =>
          console.log({ v: v.currentTarget.value, checked: v.currentTarget.checked })
        }
        id={`number-${index}`}
        className="visually-hidden"
        value={value}
      />
      <label className={className} htmlFor={`number-${index}`}>
        {value}
      </label>
      <div className="number"></div>
    </div>
  );
};

export default Number;
