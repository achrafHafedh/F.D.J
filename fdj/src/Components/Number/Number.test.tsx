import React from "react";
import { render } from "@testing-library/react";

import Number from "./Number";

describe("<Number/>", () => {
  const props = {
    index: 1,
    value: 1,
    dispatch: () => "",
    className: "big",
  };
  it("should render Number component", () => {
    const { container } = render(<Number {...props} />);
    expect(container).toMatchSnapshot();
  });
});
