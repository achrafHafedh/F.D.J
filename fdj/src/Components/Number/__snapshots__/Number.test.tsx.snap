// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Number/> should render Number component 1`] = `
<div>
  <div
    class="container-number"
  >
    <input
      class="visually-hidden"
      id="number-1"
      type="checkbox"
      value="1"
    />
    <label
      class="big"
      for="number-1"
    >
      1
    </label>
    <div
      class="number"
    />
  </div>
</div>
`;
